### PR TITLE
fix: namespaced RBACs hub api gateway

### PR DESCRIPTION
**Pull request by mmatur**
### What does this PR do?

Fix RBACs for Traefik Hub API Gateway namespaced

### Motivation

Be able to run Hub API Gateway in multiple namespace.

### More

- [x] Yes, I updated the tests accordingly
- [x] Yes, I ran `make test` and all the tests passed

### CODE REVIEW

File: traefik/templates/requirements.yaml [hunk 1]
```diff
@@ -13,6 +13,8 @@
   {{- fail "ERROR: Before traefik v3.1.0-rc3, kubernetesGateway is experimental. Enable it by setting experimental.kubernetesGateway.enabled to true" -}}
 {{- end }}
 
-{{- if and .Values.hub.token (and .Values.rbac.enabled .Values.rbac.namespaced) }}
-  {{- fail "ERROR: Currently traefik-hub doesn't support namespaced RBACs" -}}
+{{- if .Values.hub.token }}
+  {{- if and .Values.hub.apimanagement.enabled (and .Values.rbac.enabled .Values.rbac.namespaced) }}
+    {{- fail "ERROR: Currently traefik-hub doesn't support namespaced RBACs" -}}
```
Comment (mloiseleur) [inline traefik/templates/requirements.yaml:18]:
```suggestion
    {{- fail "ERROR: Currently Traefik Hub doesn't support namespaced RBACs" -}}
```

**Review by darkweaver87 — APPROVED** [review]
LGTM + cf my fix

File: traefik/templates/rbac/role.yaml [hunk 2]
```diff
@@ -31,7 +31,9 @@ rules:
     resources:
       - services
     verbs:
+      {{- if $.Values.hub.token }}
       - get
+      {{- end }}
```
Comment (mloiseleur) [inline traefik/templates/rbac/role.yaml:36]: Why adding a conditionnal here ?
In upstream 3.1 rbac [reference](https://raw.githubusercontent.com/traefik/traefik/v3.1/docs/content/reference/dynamic-configuration/kubernetes-crd-rbac.yml), this get is needed
Same with rbac [reference](https://raw.githubusercontent.com/traefik/traefik/v3.1/docs/content/reference/dynamic-configuration/kubernetes-gateway-rbac.yml) for gateway api
Sa

**Review by darkweaver87 — APPROVED** [review]
LGTM. Tested with latest version, works fine :+1:

**Pull request merged**